api/client/samples/analogous_years/lib/final_ranks_computation.py

```diff
@@ -170,14 +172,14 @@ def combined_methods_distances(dictionary_of_df):
     return rank_df
 
 
-def analogous_years(client, data_series_list, initial_date, final_date,
+def analogous_years(api_token, data_series_list, initial_date, final_date,
```

Comment:
We should think about a way to make it backward compatible, otherwise it will break any existing client side code. Maybe we could make the first argument a token_or_client_instance? thoughts?

Reply:
same for all remaining functions that the argument has changed, but this is the most important case.

Reply:
I have an idea (very wasteful).
`def analogous_years(client, data_series_list, initial_date, final_date,`...
`client = GroClient(client.api_host, client.access_token)`

Reply:
Discussed offline with John and Yiqing. Using a different hack to resolve the issue for now; so that things do not fall apart in `email_alert`.